web_widget_image_webcam/static/src/js/webcam_widget.js

```diff
@@ -0,0 +1,123 @@
+/*
+    Copyright 2016 Siddharth Bhalgami <siddharth.bhalgami@gmail.com>
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+*/
+odoo.define('web_widget_image_webcam.webcam_widget', function(require) {
+    "use strict";
+
+    var core = require('web.core');
+    var data = require('web.data');
+    var Dialog = require('web.Dialog');
+    var FieldBinaryImage = require('web.basic_fields').FieldBinaryImage;
+
+    var _t = core._t;
+    var QWeb = core.qweb;
+
+    FieldBinaryImage.include({
+
+        _render: function () {
+            this._super();
+
+            var self = this,
+                WebCamDialog = $(QWeb.render("WebCamDialog")),
+                img_data;
+
+            // ::webcamjs:: < https://github.com/jhuckaby/webcamjs >
+            // Webcam: Set Custom Parameters
+            Webcam.set({
+                width: 320,
+                height: 240,
+                dest_width: 320,
+                dest_height: 240,
+                image_format: 'jpeg',
+                jpeg_quality: 90,
+                force_flash: false,
+                fps: 45,
+                swfURL: '/web_widget_image_webcam/static/src/js/webcam.swf',
+            });
```

Comment:
it would be great if you add a options to directly pass on field and set it in the webcam.